client-mr/mr/src/main/java/org/apache/hadoop/mapred/CelebornMapOutputCollector.java

```diff
@@ -0,0 +1,129 @@
+    if ((IOBufferSize & 0x7FF) != IOBufferSize) {
+      throw new IOException("Invalid \"" + JobContext.IO_SORT_MB + "\": " + IOBufferSize);
+    }
+    jobConf.getNumReduceTasks();
+
+    CelebornConf celebornConf = HadoopUtils.fromYarnConf(jobConf);
+    JobConf celebornAppendConf = new JobConf(HadoopUtils.MR_CELEBORN_CONF);
+    String lcHost = celebornAppendConf.get(HadoopUtils.MR_CELEBORN_LC_HOST);
+    int lcPort = Integer.parseInt(celebornAppendConf.get(HadoopUtils.MR_CELEBORN_LC_PORT));
+    String applicationAttemptId = celebornAppendConf.get(HadoopUtils.MR_CELEBORN_APPLICATION_ID);
+    logger.info("Mapper initialized with celeborn {} {} {}", lcHost, lcPort, applicationAttemptId);
+    UserIdentifier userIdentifier =
+        new UserIdentifier(
+            celebornConf.quotaUserSpecificTenant(), celebornConf.quotaUserSpecificUserName());
+
+    logger.info(JobContext.IO_SORT_MB + ": " + IOBufferSize);
+    final float spiller = jobConf.getFloat(JobContext.MAP_SORT_SPILL_PERCENT, (float) 0.8);
+    int pushSize = (int) ((IOBufferSize << 20) * spiller);
+
+    SerializationFactory serializationFactory = new SerializationFactory(jobConf);
+    celebornSortBasedPusher =
+        new CelebornSortBasedPusher<>(
+            jobConf.getNumMapTasks(),
+            jobConf.getNumReduceTasks(),
+            // this is map id
+            context.getMapTask().getTaskID().getTaskID().getId(),
+            // this is attempt id
+            context.getMapTask().getTaskID().getId(),
+            serializationFactory.getSerializer(keyClass),
+            serializationFactory.getSerializer(valClass),
+            IOBufferSize << 20,
+            pushSize,
+            jobConf.getOutputKeyComparator(),
+            reporter.getCounter(TaskCounter.MAP_OUTPUT_BYTES),
+            reporter.getCounter(TaskCounter.MAP_OUTPUT_RECORDS),
+            ShuffleClient.get(applicationAttemptId, lcHost, lcPort, celebornConf, userIdentifier),
+            celebornConf);
+  }
+
+  @Override
+  public void collect(K key, V value, int partition) throws IOException {
+    reporter.progress();
+    if (key.getClass() != keyClass) {
+      throw new IOException(
+          "Type mismatch in key from map: expected "
+              + keyClass.getName()
+              + ", received "
+              + key.getClass().getName());
+    }
+    if (value.getClass() != valClass) {
+      throw new IOException(
+          "Type mismatch in value from map: expected "
+              + valClass.getName()
+              + ", received "
+              + value.getClass().getName());
+    }
+    if (partition < 0 || partition >= numReducers) {
+      throw new IOException("Illegal partition for " + key + " (" + partition + ")");
+    }
+    celebornSortBasedPusher.checkException();
+    celebornSortBasedPusher.insert(key, value, partition);
+  }
+
+  @Override
+  public void close() {
+    logger.info("Mapper collector close");
+    reporter.progress();
```

Comment:
IMO, the collector is closed and it will be cleaned after GC, I think the reporter and celebornSortBasedPusher instance will be cleaned after GC is occurred.